inference-engine/include/ie_blob.h

```diff
@@ -756,6 +824,30 @@ class TBlob : public MemoryBlob {
     void* getHandle() const noexcept override {
         return _handle;
     }
+
+    /**
+     * @brief Creates an new empty rvalue LockedMemory object.
+     *
+     * @return rvalue for the empty locked object of type T
+     */
+    virtual LockedMemory<T> data() const noexcept {
+        return std::move(lockme<T>());
+    }
+
+    /**
+     * @brief Clones Blob object of the particular class. Clone references the same memory as this object.
+     *
+     * @return Pointer to the cloned object
+     */
+    Blob* clone() const override {
+        TBlob* cloned = new TBlob<T>(this->tensorDesc, this->data(), this->byteSize());
```

Comment:
Nitpick: is `auto` usage reasonable here - looks like you already specified type on the right side of `=`?

Reply:
Not quite sure about this one. I'd prefer to have the type visible unless it is too-long-to-see